huaweicloud/services/dli/resource_huaweicloud_dli_template_flink.go

```diff
@@ -0,0 +1,279 @@
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("template.template_id", createFlinkTemplateRespBody)
+	if err != nil {
+		return diag.Errorf("error creating FlinkTemplate: ID is not found in API response")
+	}
+	d.SetId(fmt.Sprint(id))
+
+	return resourceFlinkTemplateRead(ctx, d, meta)
+}
+
+func buildCreateFlinkTemplateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":     utils.ValueIngoreEmpty(d.Get("name")),
+		"sql_body": utils.ValueIngoreEmpty(d.Get("sql")),
+		"desc":     utils.ValueIngoreEmpty(d.Get("description")),
+		"job_type": utils.ValueIngoreEmpty(d.Get("type")),
+		"tags":     utils.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
+	}
+	return bodyParams
+}
+
+func resourceFlinkTemplateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getFlinkTemplate: Query the Flink template.
+	var (
+		getFlinkTemplateHttpUrl = "v1.0/{project_id}/streaming/job-templates"
+		getFlinkTemplateProduct = "dli"
+	)
+	getFlinkTemplateClient, err := cfg.NewServiceClient(getFlinkTemplateProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DLI Client: %s", err)
+	}
+
+	getFlinkTemplatePath := getFlinkTemplateClient.Endpoint + getFlinkTemplateHttpUrl
+	getFlinkTemplatePath = strings.ReplaceAll(getFlinkTemplatePath, "{project_id}", getFlinkTemplateClient.ProjectID)
+
+	getFlinkTemplatequeryParams := buildGetFlinkTemplateQueryParams(d)
+	getFlinkTemplatePath += getFlinkTemplatequeryParams
+
+	getFlinkTemplateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getFlinkTemplateResp, err := getFlinkTemplateClient.Request("GET", getFlinkTemplatePath, &getFlinkTemplateOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving FlinkTemplate")
+	}
+
+	getFlinkTemplateRespBody, err := utils.FlattenResponse(getFlinkTemplateResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jsonPath := fmt.Sprintf("template_list.templates[?template_id==`%s`]|[0]", d.Id())
+	flinkTemplate := utils.PathSearch(jsonPath, getFlinkTemplateRespBody, nil)
+	if flinkTemplate == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "no data found")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", flinkTemplate, nil)),
+		d.Set("sql", utils.PathSearch("sql_body", flinkTemplate, nil)),
+		d.Set("description", utils.PathSearch("desc", flinkTemplate, nil)),
+		d.Set("type", utils.PathSearch("job_type", flinkTemplate, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetFlinkTemplateQueryParams(d *schema.ResourceData) string {
+	res := "&limit=100"
+	if v, ok := d.GetOk("name"); ok {
+		res = fmt.Sprintf("%s&name=%v", res, v)
+	}
+
+	if res != "" {
```

Comment:
it is unnecessary to check res, it is not empty forever

Reply:
done